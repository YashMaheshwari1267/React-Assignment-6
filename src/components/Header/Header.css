li {
    list-style: none;
    display: inline;
    float: left;
    width: auto;
}

ul {
    background-color: #34cceb;
    list-style-type: none;
    margin: 0;
    padding: 0;
    overflow: hidden;
}

a {
    display: block;
    background-color: #34cceb;
    font-weight: bold;
    color: black;
}

li a {
    display: block;
    color: white;
    text-align: center;
    padding: 14px 16px; 
    text-decoration: none;
  }
  
li a:hover {
    background-color: #c7326b;
}

a:link {
    background-color: #34cceb;
}

button {
    background-color: antiquewhite;
    color: red;
    font-weight: bold;
    font-size: large;
    box-sizing: content-box;
    block-size: 40px;
}