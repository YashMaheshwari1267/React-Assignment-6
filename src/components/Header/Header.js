import React from 'react';
import { Route } from 'react-router-dom';
import './Header.css';

const Header = () => {
    return (
    <div className="Header">    
      <ul>
        <li><a href="/">HOME</a></li>
        <li><a href="/projects">PROJECTS</a></li>
        <li><a href="/services">SERVICES</a></li>
        <li><a href="/contact">CONTACT</a></li>
      </ul>
      <Route path='/' exact render={() => <h1>You Chose <button>HOME</button></h1>} />
      <Route path='/projects' exact render={() => <h1>You Chose <button>PROJECTS</button></h1>} />
      <Route path='/services' exact render={() => <h1>You Chose <button>SERVICES</button></h1>} />
      <Route path='/contact' exact render={() => <h1>You Chose <button>CONTACT</button></h1>} />
    </div>
    )
}

export default Header;