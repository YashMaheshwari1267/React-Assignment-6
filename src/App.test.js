import React from 'react';
import { mount, shallow } from 'enzyme';
import App from './App';
import Header from './components/Header/Header';

describe('rendering ',() => {
  it('should render the <App /> component',() => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  })

  it('should render ul',() => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find("ul").length).toEqual(1);
  })

  it('should render li',() => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find("li").length).toEqual(4);
  })

})

describe('macthing elements',() => {
  it('list heading should match',() => {
    const wrapper = shallow(<Header />);
    let list = wrapper.find("li");
    expect(list.first().text()).toEqual("HOME");
    expect(list.at(1).text()).toEqual("PROJECTS"); 
    expect(list.at(2).text()).toEqual("SERVICES"); 
    expect(list.at(3).text()).toEqual("CONTACT");  
  })
})