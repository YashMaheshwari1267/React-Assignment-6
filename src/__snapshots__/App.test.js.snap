// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering  should render the <App /> component 1`] = `ShallowWrapper {}`;

exports[`rendering components should render the <App /> component 1`] = `ShallowWrapper {}`;
